.nginx/nginx.conf

```diff
@@ -0,0 +1,15 @@
+worker_processes 4;
```

Comment:
Is there a reason you chose 4 worker processes?

Reply:
No particular reason. In fact it may be ideal to set it to `auto` instead in a later update.

Reply:
When reading about worker nodes (the documentation), I read about manager nodes:

> Worker nodes don’t participate in the Raft distributed state, make scheduling decisions, or serve the swarm mode HTTP API.

The above being processes of the manager node. However, _the manager node is a worker node as well._ 
**Where do we differentiate between manager nodes and worker nodes? How is this declared, or instantiated so to speak?**

Reply:
I know we need at least 1 manager node in order to have worker nodes. I'm guessing we have 1 manager node in this system; Why do we have 1 manager node, rather than 2? Where would such a system be implemented (By that I mean using more than 1 manager node)?

Reply:
> Where do we differentiate between manager nodes and worker nodes? How is this declared, or instantiated so to speak?

All happens under the hood. Nginx takes care of it unless we want to tinker around with the low level configurations

> Why do we have 1 manager node, rather than 2? Where would such a system be implemented (By that I mean using more than 1 manager node)?

Having 1 suffices for our needs. Unless its in an enterprise context with millions of active visitors we wouldn't need to scale beyond that. I've personally never had to work with in a situation that needed something like that setup. That being said there are certainly situations outside the scope of our development here that would use that type of setup. The point of having multiple nodes in an orchestration setting is usually for 2 reasons --> fast scalability and fault tolerance (healing & auto-provisioning). Locally I've worked with multiple nodes in a Docker Swarm setting (which we're not using for the portal because I opted for Kubernetes instead). The architectural scenarios that require multiple manager nodes would be HA clustering (high availability). When creating a network or cluster that you want especially fault tolerant there are Active-Active / Active-Passive HA Clustering techniques that can leverage the benefits of multiple nodes. Might be interesting to read up on but is not something that we need for our application.